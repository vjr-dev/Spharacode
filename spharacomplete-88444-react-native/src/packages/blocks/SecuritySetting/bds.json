{
  "bds": {
    "name": "SecuritySetting",
    "id": "SecuritySetting",
    "platform": [
      "desktop",
      "mobile"
    ],
    "author": "CMT",
    "summary": null,
    "license": "BSD",
    "version": "1.0.0",
    "pages": [
      {
        "MoreInSocial": {
          "hotspots": [
          ]
        }
      }
    ]
  }
}
