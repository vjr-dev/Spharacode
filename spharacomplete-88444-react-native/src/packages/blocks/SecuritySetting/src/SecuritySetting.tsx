// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import {
  ActivityIndicator,
  Image,
  ImageBackground,
  Modal,
  Platform,
  SafeAreaView,
  StatusBar,
  Switch,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { RFValue } from "react-native-responsive-fontsize";
import { COLORS } from "../../../framework/src/Globals";
import { back1, back2, image_back } from "./assets";
import SecuritySettingController, { Props } from "./SecuritySettingController";
import { Styles } from "./SecuritySettingStyle";

export default class SecuritySetting extends SecuritySettingController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <SafeAreaView style={Styles.container1}>
        <View style={Styles.child}>
          <StatusBar
            animated={true}
            backgroundColor="#454545"
            barStyle={"light-content"}
            showHideTransition={"slide"}
            // hidden={true}
          />
          <ImageBackground source={back1} style={Styles.image}>
            <ImageBackground
              source={back2}
              style={[Styles.image, { alignItems: "center" }]}
            >
              <View style={Styles.view1}>
                <TouchableOpacity
                  testID="backButton"
                  style={Styles.button1}
                  onPress={() => this.goback()}
                >
                  <Image source={image_back} style={Styles.backbutton} />
                </TouchableOpacity>
              </View>

              <Text style={Styles.security}>Security.</Text>
              <Text style={Styles.description}>
                When 2-step is tuned ON.you will prompted for your passcode when
                you are trying to re-login or restore your profile
              </Text>

              <View style={Styles.switchView}>
                <Text style={Styles.switchText}>2-step Verification</Text>
                <Switch
                  testID="twoStepSwitch"
                  trackColor={{ false: COLORS.black, true: COLORS.darkorange }}
                  thumbColor={COLORS.white}
                  ios_backgroundColor="#000"
                  onValueChange={(Vala) => this.switch11(Vala)}
                  value={this.state.switch1}
                  style={Styles.switchSize}
                  // style={{ backgroundColor: "red" }}
                />
              </View>
              {this.state.switch1 ? (
                <>
                  <View
                    style={[
                      Styles.optionView,
                      { marginTop: Platform.OS === "android" ? 40 : 50 },
                    ]}
                  >
                    <Text style={Styles.optionText}>Edit Passcode</Text>
                    <TouchableOpacity
                      testID="editPasscodeButton"
                      onPress={() => this.Editbuttonclick()}
                    >
                      <Text style={[Styles.optionAction, { marginRight: 15 }]}>
                        EDIT
                      </Text>
                    </TouchableOpacity>
                  </View>

                  <View style={[Styles.optionView, { marginTop: 35 }]}>
                    <Text style={Styles.optionText}>Update Email</Text>
                    <TouchableOpacity
                      testID="editEmailButton"
                      onPress={() => this.Updatebuttonclick()}
                    >
                      <Text style={Styles.optionAction}>UPDATE</Text>
                    </TouchableOpacity>
                  </View>
                </>
              ) : null}

              <Modal visible={this.state.Modal1} transparent={true}>
                <View style={Styles.modalView}>
                  <View style={Styles.modalCenteredView}>
                    <View style={Styles.alertMessageView}>
                      <Text style={Styles.alertMessage}>
                        Do you really want to disable the 2-step verification?{" "}
                      </Text>
                    </View>
                    <View style={Styles.seperator} />
                    <View style={Styles.alertBox}>
                      <View style={Styles.alertButtonView}>
                        <TouchableOpacity
                          testID="noButton"
                          style={{ width: 50 }}
                          onPress={() =>
                            this.setState({ Modal1: false, switch1: true })
                          }
                        >
                          <Text style={Styles.noText}>NO</Text>
                        </TouchableOpacity>
                        <TouchableOpacity
                          testID="yesButton"
                          style={{ width: 50 }}
                          onPress={() =>
                            this.setState({ Modal1: false, switch1: false })
                          }
                        >
                          <Text
                            style={[Styles.noText, { color: COLORS.black }]}
                          >
                            YES
                          </Text>
                        </TouchableOpacity>
                      </View>
                    </View>
                    {/* <View style={Styles.alertButtonView}>
                                            <TouchableOpacity
                                                style={{alignSelf:"flex-end",backgroundColor:"yellow"}}
                                                onPress={() => this.setState({ Modal1: false, switch1: true })}
                                            >
                                                <Text style={Styles.noText}>NO</Text>
                                            </TouchableOpacity>
                                            <TouchableOpacity onPress={() => this.setState({ Modal1: false, switch1: false })}>
                                                <Text style={{ fontSize: 17 }}>YES</Text>
                                            </TouchableOpacity>
                                        </View> */}
                  </View>
                </View>
              </Modal>

              <Modal visible={this.state.Loader} transparent={true}>
                <View style={Styles.modalView}>
                  <View
                    style={[Styles.modalCenteredView, { height: RFValue(100) }]}
                  >
                    <ActivityIndicator
                      animating={true}
                      size={"large"}
                      color="#f07233"
                    />
                  </View>
                </View>
              </Modal>
            </ImageBackground>
          </ImageBackground>
        </View>
      </SafeAreaView>
    );
  }
}
// Customizable Area End
