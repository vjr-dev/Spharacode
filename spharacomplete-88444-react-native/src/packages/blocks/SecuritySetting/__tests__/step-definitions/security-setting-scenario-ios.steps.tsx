//@ts-nocheck
//@ts-ignore
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import SecuritySetting from "../../src/SecuritySetting";
import { Platform } from "react-native";
jest.useFakeTimers();
const screenProps = {
  navigation: {
    dispatch: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  id: "SecuritySetting",
};

const feature = loadFeature(
  "./__tests__/features/security-setting-scenario.feature"
);

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.mock("react-native/Libraries/Utilities/Platform", () => ({
      OS: "ios",
      select: () => null,
    }));
  });

  test("User navigates to SecuritySetting screen", ({ given, when, then }) => {
    let securitySettingWrapper: ShallowWrapper;
    let instance: SecuritySetting;

    given("I am a User attempting to SecuritySetting Screen", () => {
      securitySettingWrapper = shallow(<SecuritySetting {...screenProps} />);
      expect(securitySettingWrapper).toBeTruthy();
      expect(securitySettingWrapper).toMatchSnapshot();
    });

    when("I navigate to the SecuritySetting Screen", () => {
      instance = securitySettingWrapper.instance() as SecuritySetting;
      expect(instance).toBeTruthy();
    });

    then("I can press back button", async () => {
      Platform.OS = "android";
      let backButton = securitySettingWrapper.findWhere(
        (node) => node.prop("testID") === "backButton"
      );
      backButton.simulate("press");
      expect(backButton).toBeTruthy();
    });

    then("I can turn on or off two step verifications", async () => {
      Platform.OS = "ios";
      let twoStepSwitch = securitySettingWrapper.findWhere(
        (node) => node.prop("testID") === "twoStepSwitch"
      );
      twoStepSwitch.props().onValueChange(true);
      twoStepSwitch.props().onValueChange(false);
      expect(twoStepSwitch).toBeTruthy();
    });

    then("I can edit passcode", async () => {
      let editPasscodeButton = securitySettingWrapper.findWhere(
        (node) => node.prop("testID") === "editPasscodeButton"
      );
      editPasscodeButton.simulate("press");
      expect(editPasscodeButton).toBeTruthy();
    });

    then("I can edit email", async () => {
      let editEmailButton = securitySettingWrapper.findWhere(
        (node) => node.prop("testID") === "editEmailButton"
      );
      editEmailButton.simulate("press");
      expect(editEmailButton).toBeTruthy();
    });

    then("I can accept two step verification on modal", async () => {
      let yesButton = securitySettingWrapper.findWhere(
        (node) => node.prop("testID") === "yesButton"
      );
      yesButton.simulate("press");
      expect(yesButton).toBeTruthy();
    });

    then("I can decline two step verification on modal", async () => {
      let noButton = securitySettingWrapper.findWhere(
        (node) => node.prop("testID") === "noButton"
      );
      noButton.simulate("press");
      expect(noButton).toBeTruthy();
    });
  });
});
