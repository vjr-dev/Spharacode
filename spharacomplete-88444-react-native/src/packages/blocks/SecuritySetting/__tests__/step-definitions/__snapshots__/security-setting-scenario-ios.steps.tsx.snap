// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecuritySetting User navigates to SecuritySetting screen 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <StatusBar
      animated={true}
      backgroundColor="#454545"
      barStyle="light-content"
      showHideTransition="slide"
    />
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/SecuritySetting/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/SecuritySetting/assets/image_background-2.png",
          }
        }
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "height": 60,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "marginLeft": 20,
              }
            }
            testID="backButton"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/SecuritySetting/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 15,
                  "tintColor": "#f07233",
                  "width": 15,
                }
              }
            />
          </TouchableOpacity>
        </View>
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontSize": 20,
              "marginTop": 40,
              "width": 725,
            }
          }
        >
          Security.
        </Text>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "fontSize": 16,
              "marginTop": 10,
              "width": 725,
            }
          }
        >
          When 2-step is tuned ON.you will prompted for your passcode when you are trying to re-login or restore your profile
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "height": 50,
              "marginTop": 40,
              "width": 725,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255,255,255)",
                "fontSize": 20,
                "width": "85%",
              }
            }
          >
            2-step Verification
          </Text>
          <Switch
            ios_backgroundColor="#000"
            onValueChange={[Function]}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scaleX": 0.9,
                  },
                  Object {
                    "scaleY": 0.9,
                  },
                ],
              }
            }
            testID="twoStepSwitch"
            thumbColor="rgb(255,255,255)"
            trackColor={
              Object {
                "false": "rgb(0,0,0)",
                "true": "rgba(255,138,73,1)",
              }
            }
            value={false}
          />
        </View>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#fff",
                  "height": 334,
                  "width": 613,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "width": 573,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "rgb(0,0,0)",
                      "fontSize": 31,
                      "paddingBottom": 50,
                      "paddingTop": 30,
                    }
                  }
                >
                  Do you really want to disable the 2-step verification?
                   
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#a1a1a1",
                    "height": 1,
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginRight": 20,
                      "width": 110,
                    }
                  }
                >
                  <TouchableOpacity
                    activeOpacity={0.2}
                    onPress={[Function]}
                    style={
                      Object {
                        "width": 50,
                      }
                    }
                    testID="noButton"
                  >
                    <Text
                      style={
                        Object {
                          "color": "#a1a1a1",
                          "fontSize": 15,
                        }
                      }
                    >
                      NO
                    </Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    activeOpacity={0.2}
                    onPress={[Function]}
                    style={
                      Object {
                        "width": 50,
                      }
                    }
                    testID="yesButton"
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#a1a1a1",
                            "fontSize": 15,
                          },
                          Object {
                            "color": "rgb(0,0,0)",
                          },
                        ]
                      }
                    >
                      YES
                    </Text>
                  </TouchableOpacity>
                </View>
              </View>
            </View>
          </View>
        </Component>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#fff",
                    "height": 334,
                    "width": 613,
                  },
                  Object {
                    "height": 196,
                  },
                ]
              }
            >
              <ActivityIndicator
                animating={true}
                color="#f07233"
                hidesWhenStopped={true}
                size="large"
              />
            </View>
          </View>
        </Component>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;
